packages/react-dom-bindings/src/events/plugins/SimpleEventPlugin.js

```diff
@@ -219,6 +220,16 @@ function extractEvents(
         nativeEvent,
         nativeEventTarget,
       );
+      const isAtTarget = event.eventPhase === 2;
+      // native cancel event should only fired on AT_TARGET
+      // https://developer.mozilla.org/en-US/docs/Web/API/Event/eventPhase
+      if (
+        domEventName === 'cancel' &&
+        (eventSystemFlags & IS_NON_DELEGATED) !== 0 &&
+        !isAtTarget
+      ) {
+        return;
+      }
```

Comment:
I don't think this is quite the right fix. In fact we do want the input cancel event to bubble as that's what would happen in the native browser event system and we are looking to move closer to that over time.

But it shouldn't double-trigger the higher listener. The reason that happens is that React has native listeners on the input itself as well as on the dialog; when the bubbling native event is dispatched, it triggers both listeners, and React then bubbles each with its own event system. Really I think the correct fix is probably to change listenToNonDelegatedEvent so that it only responds to events whose target is equal to the element we are adding the listener on (equivalently, ones where event.target === event.currentTarget when the native listener is fired). The intent of that function (unless I'm mistaken) is only to catch events of that kind. We should verify that all current uses of that are not meant to catch any bubbled events though.

Reply:
At first, I also wanted to fix it according to this idea (validate `target === currentTarget`). But I found that in the original HTML, the cancel event can bubble and be fired when `target !== currentTarget`(This seems to be  inconsistent with the standard defined by MDN). Therefore, validating 
 `target === currentTarget` can't solve this problem, my current solution is not quite correct either. Due to the inconsistency between the actual behavior of the browser and the standard defined by MDN, should we consider removing it from nonDelegatedEvents so that it works correctly?

Here's the codesandbox: https://codesandbox.io/p/sandbox/reverent-margulis-gvgwq2?file=%2Findex.html%3A14%2C72

<img width="1725" alt="image" src="https://github.com/facebook/react/assets/73059627/387f7181-2090-4f6c-b7a5-d8556102e0cc">

Reply:
If I'm not mistaken, the issue is that for the dialog cancel event it does not bubble.

Reply:
> If I'm not mistaken, the issue is that for the dialog cancel event it does not bubble.

Yes, according to the standard defined by MDN，the dialog cancel event should not bubble, but in testing with native HTML in the browser, the result is that it does bubble. So, we should decide whether to modify it based on the actual behavior of the browser or adhere to the standard defined by MDN.

Reply:
Are you sure? I'm having trouble reproing the event at all but in the spec it's not mentioned as bubbling https://html.spec.whatwg.org/multipage/interactive-elements.html and in Blink source I don't see bubbling either. https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/html/html_dialog_element.cc;drc=67d90538f11c6b232dbfd716075db52aeb34fd15;l=346?q=showModal%20lang:cpp%20filepath:third_party%2Fblink&ss=chromium

If the spec and browsers are inconsistent then we should file a bug with them; if MDN is inconsistent with the spec and browsers then we should file a bug with MDN.